docs/apps/components/number.md

```diff
@@ -0,0 +1,193 @@
+# Number
+
+## Description
+
+**Number** component is a UI element that allows users to input and manipulate numerical values within a form.
+
+It's commonly used for collecting quantitative data, such as age, quantity, price, or measurement values.
+
+<img src= "/apps/components/img/number.png">
+
+## Use Case
+
+1. Text Input with Number Type
+2. Number Spinner
+3. Range Slider
+4. Number Validation
+5. Custom Number Input
+
+Examples mat include:
+
+1. Collecting Age
+2. Quantity Selection
+3. Price Entry
+4. Parameter Adjustment
+5. Measurement Input
+
+### Interactivity
+
+1. **Real-time Updates**: For instance, as the user changes the quantity in a product order form, the corresponding subtotal or total price can be updated dynamically.
+
+2. **Conditional Logic**: For example, if the user enters a specific age range in a registration form, additional questions or fields relevant to that age group can be shown.
+
+3. **Interactive Calculations**:For instance, in a loan application form, the user can enter the loan amount, interest rate, and repayment period, and the component can dynamically calculate the monthly installment amount.
+
+4. **Data Validation**: For example, in a form collecting age, validate that the input is a positive integer within a reasonable age range.
+
+5. **Visual Enhancements**: For example, create custom input fields with color changes or animations based on input values or provide interactive sliders or spinners for precise value adjustments.
+
+## API
+
+### Events
+
+| **Name**| **Description**|
+|---------|----------------|
+| `onChange (element)`| The onChange event occurs when the value of a field is changed|
+| `onFocus (element)`| The onfocus event occurs when a field gets focus|
+| `onBlur (element)`| The onblur event occurs when a field loses focus|
+
+### Methods
+
+| **Name**| **Description**|**Parameters**|
+|---------|----------------|--------------|
+|`this.addClassName`|Add style class to a form item|`(fields: String`&#124;`String[], className: String)`|
+|`this.removeClassName`|Remove style class from a form item|`(fields: String`&#124;`String[], className: String)`|
+|`this.getComponent`|Returns a component whose id has been passed as a parameter|`(component_ID: String) : Object)`|
+|`this.getValues`|Gets the current values of all fields|`() : Object`|
+|`this.hide`|Hides the field|`(fields: String`&#124;`String[])`|
+|`this.show`|Displays the field|`(fields: String`&#124;`String[])`|
+|`this.disable`| Disable number field from user interaction|`(fields: String`&#124;`String[])`|
+|`this.enable`| Enable number field from user interaction|`(fields: String`&#124;`String[])`|
+|`this.getValue`|Get A Value From a component|`(fieldName: String)`|
+|`this.setData`|Set the data in the field|`(Value: Object)`|
```

Comment:
Please change the description so that is shows an example of the structure of the Value Object. Like this: ** Set the data in the field. The `Value` object should be of type `{ componentId: componentValue }` **

Reply:
done